### PR TITLE
[flutter_tools] Update WebAssetServer to avoid context, fix tests

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

WebAssetServer could fail if some of the global statics it depended on were initialized in a different order. Fix this by removing globals.

Delete dwds startup test that needs to spawn a real server.

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 1]
```diff
@@ -817,36 +824,50 @@ class WebDevFS implements DevFS {
 }
 
 class ReleaseAssetServer {
-  ReleaseAssetServer(this.entrypoint);
+  ReleaseAssetServer(this.entrypoint, {
+    @required FileSystem fileSystem,
+    @required String webBuilDirectory,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:829]:
```suggestion
    @required String webBuildDirectory,
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 2]
```diff
@@ -173,7 +174,13 @@ class WebAssetServer implements AssetReader {
 
       // In release builds deploy a simpler proxy server.
       if (buildMode != BuildMode.debug) {
-        final ReleaseAssetServer releaseAssetServer = ReleaseAssetServer(entrypoint);
+        final ReleaseAssetServer releaseAssetServer = ReleaseAssetServer(
+          entrypoint,
+          fileSystem: globals.fs,
+          platform: globals.platform,
+          flutterRoot: Cache.flutterRoot,
+          webBuilDirectory: getWebBuildDirectory(),
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:182]:
```suggestion
          webBuildDirectory: getWebBuildDirectory(),
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 3]
```diff
@@ -817,36 +824,50 @@ class WebDevFS implements DevFS {
...
-  ReleaseAssetServer(this.entrypoint);
+  ReleaseAssetServer(this.entrypoint, {
+    @required FileSystem fileSystem,
+    @required String webBuilDirectory,
+    @required String flutterRoot,
+    @required Platform platform,
+  }) : _fileSystem = fileSystem,
+       _platform = platform,
+       _flutterRoot = flutterRoot,
+       _webBuilDirectory = webBuilDirectory,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:835]:
```suggestion
       _webBuildDirectory = webBuildDirectory,
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 4]
```diff
@@ -817,36 +824,50 @@ class WebDevFS implements DevFS {
...
+       _fileSystemUtils = FileSystemUtils(fileSystem: fileSystem, platform: platform);
 
   final Uri entrypoint;
+  final String _flutterRoot;
+  final String _webBuilDirectory;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:840]:
```suggestion
  final String _webBuildDirectory;
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 5]
```diff
@@ -817,36 +824,50 @@ class WebDevFS implements DevFS {
...
 
   // Locations where source files, assets, or source maps may be located.
-  final List<Uri> _searchPaths = <Uri>[
-    globals.fs.directory(getWebBuildDirectory()).uri,
-    globals.fs.directory(Cache.flutterRoot).uri,
-    globals.fs.directory(Cache.flutterRoot).parent.uri,
-    globals.fs.currentDirectory.uri,
-    globals.fs.directory(globals.fsUtils.homeDirPath).uri,
+  List<Uri> _searchPaths() => <Uri>[
+    _fileSystem.directory(_webBuilDirectory).uri,
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:847]:
```suggestion
    _fileSystem.directory(_webBuildDirectory).uri,
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart [hunk 6]
```diff
@@ -857,7 +878,7 @@ class ReleaseAssetServer {
       });
     }
     if (request.url.path == '') {
-      final File file = globals.fs.file(globals.fs.path.join(getWebBuildDirectory(), 'index.html'));
+      final File file = _fileSystem.file(_fileSystem.path.join(_webBuilDirectory, 'index.html'));
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/build_runner/devfs_web.dart:881]:
```suggestion
      final File file = _fileSystem.file(_fileSystem.path.join(_webBuildDirectory, 'index.html'));
```

File: packages/flutter_tools/lib/src/build_runner/devfs_web.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/devfs_web.dart:829]: D'oh

File: packages/flutter_tools/test/general.shard/web/web_asset_server_test.dart [hunk 7]
```diff
@@ -93,39 +99,37 @@ void main() {
...
       ..createSync(recursive: true)
       ..writeAsStringSync('void main() { }');
     final Response response = await assetServer
       .handle(Request('GET', Uri.parse('http://localhost:8080/bar.dart')));
 
     expect(response.statusCode, HttpStatus.ok);
-  }, overrides: <Type, Generator>{
-    FileSystem: () => fileSystem,
-    Platform: () => platform,
-    ProcessManager: () => FakeProcessManager.any(),
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/web/web_asset_server_test.dart:129]: was the process manager never getting called?

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/web/web_asset_server_test.dart:129]: I guess not, since these are now `testWithoutContext`. cool.

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/web/web_asset_server_test.dart:129]: Negatory, but we have a rule that a context injected FIleSystem requires a context Injected ProcessManager

**Review by christopherfujino — APPROVED** [review]
LGTM once the analyzer verifies all the spelling changes are consistent.

**Comment by jonahwilliams** [comment]
Almost! Sorry I didn't accept the suggestions, that seems to trip up the CLA bot

**Comment by christopherfujino** [comment]
> Almost! Sorry I didn't accept the suggestions, that seems to trip up the CLA bot

Not offended at all. It's probably easier to do locally anyway.

**Comment by flutter-github-sync** [comment]
Started Google testing for this PR

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Pull request merged**